src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -135,104 +137,89 @@ private static SpanContext ExtractPropagatedContext(HttpRequest request)
             return Enumerable.Empty<KeyValuePair<string, string>>();
         }
 
-        private bool ShouldIgnore(HttpContext httpContext)
-        {
-            foreach (Func<HttpContext, bool> ignore in _options.IgnorePatterns)
-            {
-                if (ignore(httpContext))
-                {
-                    return true;
-                }
-            }
-
-            return false;
-        }
-
         private void OnHostingHttpRequestInStart(object arg)
         {
-            var httpContext = HttpRequestInStartHttpContextFetcher.Fetch<HttpContext>(arg);
+            var tracer = _tracer ?? Tracer.Instance;
 
-            if (ShouldIgnore(httpContext))
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationName))
```

Comment:
That's probably a good candidate for optimization, I'll have to measure

Reply:
Indeed. As long as the `Tracer.Instance` reference is the same, I don't think `tracer.Settings.IsIntegrationEnabled(IntegrationName)` can change. This is called from all integrations, too.